tests/platform_tests/api/test_component.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import re
+
+import pytest
+import yaml
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.platform_api import chassis, component
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.disable_loganalyzer,  # disable automatic loganalyzer
+    pytest.mark.topology('any')
+]
+
+image_list = [
+    "current",
+    "next"
+]
+
+class TestComponentApi(object):
```

Comment:
Suggest to inherit from `PlatformApiTestBase` and utilize the `expect` function in order test all components before asserting.